advisor_engine/endpoints.py

```diff
@@ -117,8 +125,26 @@ def handle_api_ping():
     return Response(status_code=status.HTTP_200_OK)
 
 
+class HeaderSizeLimitMiddleware(BaseHTTPMiddleware):
```

Comment:
Doesn't something like this exist on the web framework or in some library?

Reply:
In FastAPI applications, this has to be handled at the underlying ASGI layer. In this case Uvicorn, which does not appear to support this. So I believe this has to be implemented as middleware in FastAPI as it does not have any native config options built in to handle limiting the number of headers and their size. Additionally there do not seem to be any built-in options for uvicorn to handle this.

- Uvicorn settings - [https://www.uvicorn.org/settings/#resource-limits](https://www.uvicorn.org/settings/#resource-limits)
- FastAPI class reference - [https://fastapi.tiangolo.com/reference/fastapi/#fastapi.FastAPI](https://fastapi.tiangolo.com/reference/fastapi/#fastapi.FastAPI)

Reply:
Thanks for the explanation. Having this on the webserver (CGI layer) would be the best choice. Should more our services start to use uvicorn, the middleware layer might be made as a library or contributed to Uvicorn.